File: app/src/main/java/net/daverix/ajvm/jvm/GotoOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class GotoOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int gotoOffset = reader.readUnsignedShort();
        reader.jumpTo(indexOfBytecode + gotoOffset);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/GotoOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class GotoOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val gotoOffset = reader.readUnsignedShort()
        reader.jumpTo(indexOfBytecode + gotoOffset)
    }
}
